packages/host/app/services/realm-server.ts

```diff
@@ -177,25 +178,37 @@ export default class RealmServerService extends Service {
   }
 
   async fetchCreditInfo(): Promise<CreditInfo> {
-    let response = await this.network.authedFetch(`${this.url.origin}/_user`);
+    if (!this.client) {
+      throw new Error(`Cannot fetch credit info without matrix client`);
+    }
+    await this.login();
+    if (this.auth.type !== 'logged-in') {
+      throw new Error('Could not login to realm server');
+    }
+
+    let response = await this.network.authedFetch(`${this.url.origin}/_user`, {
+      headers: {
+        Accept: SupportedMimeType.JSONAPI,
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${this.token}`,
```

Comment:
I’m surprised to see this header as I’d assume `authedFetch` would include something similar, looking at the code it’s not clear to me that it does though, so I’m just confused!

Reply:
There are two types of authentication: realm authentication and realm server authentication. The `authedFetch` function is used to handle realm authentication. I think using `fetch` here would make it less confusing.

Reply:
Updated [c25235d](https://github.com/cardstack/boxel/pull/1756/commits/c25235d6563553f81801950a1cbca87a2dec55a9).